Submitted By: Alexander E. Patrakov
Date: 2005-08-12
Initial Package Version: 2.86
Upstream Status: LiveCD specific
Origin: Alexander E. Patrakov
Description: Makes the "init u" command work even if the system is already in
runlevel 0 or 6. This is needed in order to free the root filesystem.

diff -ur sysvinit-2.86.orig/src/init.c sysvinit-2.86/src/init.c
--- sysvinit-2.86.orig/src/init.c	2004-07-30 18:16:20.000000000 +0600
+++ sysvinit-2.86/src/init.c	2005-06-08 14:57:03.000000000 +0600
@@ -1835,9 +1835,6 @@
 	char		**env;
 	int		fd;
 
-	if (strchr("S12345",runlevel) == NULL)
-		return;
-
 	/*
 	 *	Reset the alarm, and block all signals.
 	 */
